agent/reconcile.go

```diff
@@ -142,16 +142,25 @@ func desiredAgentState(a *Agent, reg registry.Registry) (*AgentState, error) {
 	for _, u := range units {
 		u := u
 		md := u.RequiredTargetMetadata()
-		if u.IsGlobal() && !machine.HasMetadata(&ms, md) {
```

Comment:
@ewoutp Exactly. It doesn't make any difference.
Why did it change? Short answer is, there's no particular reason.

Long answer is like that: @wuqixuan wrote the 1st patch ("fleetd: support conflict in global unit") by splitting the loop into 2 loops. However, @jonboulle [suggested](https://github.com/coreos/fleet/pull/1270#discussion_r40958036) keeping a single loop. Unfortunately the original author @wuqixuan didn't respond, so I took it over, and wrote the 3rd patch ("agent: squash loops into a single one in desiredAgentState") to follow @jonboulle's suggestion. The patch effectively changed it back to a single loop, but code style in that particular part changed slightly. Thus now you see the difference in a unified diff view.

I suppose this difference is not critical for the entire functionality. But if you want me to change it again, sure, I can update the patch again.